core/api_v2.go

```diff
@@ -505,21 +505,35 @@ func (c *Core) v2API() *route.Router {
 		}
 
 		log.Infof("registering message bus web client")
-		ch, _, err := c.bus.Register(queues)
+		ch, slot, err := c.bus.Register(queues)
 		if err != nil {
 			r.Fail(route.Oops(err, "Unable to begin streaming SHIELD events"))
 			return
 		}
+		log.Infof("registered with message bus as [slot:%d]", slot)
 
-		go socket.Discard()
+		go socket.Discard(func() {
+			defer recover()
+			close(ch)
```

Comment:
closing this channel here is redundant, as it is also closed in Unregister. Closing a closed channel will result in a panic.

Reply:
However, it is necessary to break out of the loop. We probably need to do a non-blocking read in Unregister to determine if the channel has already been closed.

Reply:
I found the “ignore panic” of the `defer recover()` works well.

Reply:
Yes, but this isn't the thread that is going to panic. This will fire first if the client closes the websocket, and then Unregister will call it the second time and then panic.

Reply:
Other option might be to `close(ch)` in the `done` branch of `socket.Write(b)` (new file line 524), right before we break out of the loop, and then _not_ close the channel in Unregister;  I'm unclear if anyone else is calling Unregister out of band or not.

Reply:
Belt, meet suspenders. bc9f73a has a fix that is cautiously safe (I think).